BHoMUpgrader41/Converter.cs

```diff
@@ -60,6 +65,100 @@ public Converter() : base()
             return newVersion;
         }
 
+        public static Dictionary<string, object> UpgradeHarpoonSettings(Dictionary<string, object> harpoonSettings)
+        {
+            if (harpoonSettings == null)
+                return null;
+
+            Dictionary<string, object> newHarpoonSettings = new Dictionary<string, object>(harpoonSettings);
+            if (!newHarpoonSettings.ContainsKey("AdvanceConfig"))
+            {
+                newHarpoonSettings.Add("AdvanceConfig", new List<string>());
+            }
+
+            return newHarpoonSettings;
+        }
+
+        public static Dictionary<string, object> UpgradeRadiantSource(Dictionary<string, object> radiantSource)
+        {
+            if (radiantSource == null)
+                return null;
+
+            Dictionary<string, object> energySource = new Dictionary<string, object>(radiantSource);
+            energySource["_t"] = energySource["_t"].ToString().Replace("RadiantSource", "EnergySource");
+
+            if (!energySource.ContainsKey("EnergyFluxInput"))
+            {
+                energySource.Add("EnergyFluxInput", "0 [W m^-2]");
+            }
+            if (energySource.ContainsKey("Direction"))
+            {
+                energySource.Add("RadiationDirection", radiantSource["Direction"]);
+                energySource.Remove("Direction");
+            }
+
+            return energySource;
+        }
+
+        public static Dictionary<string, object> UpgradeAdiabaticBoundary(Dictionary<string, object> adiabaticBoundary)
+        {
+            if (adiabaticBoundary == null)
+                return null;
+
+            Dictionary<string, object> newAdiabaticBoundary = new Dictionary<string, object>(adiabaticBoundary);
+            if (newAdiabaticBoundary.ContainsKey("RadiantSource"))
+            {
+                newAdiabaticBoundary.Add("EnergySource", newAdiabaticBoundary["RadiantSource"]);
+                newAdiabaticBoundary.Remove("RadiantSource");
+            }
```

Comment:
Property rename doesn't require a custom upgrader.

See https://github.com/BHoM/documentation/wiki/Versioning---How-to-modify-code-without-breaking-user-scripts#modifying-property-names for more details